model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,251 @@
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node
+from mo.graph.perm_inputs import PermuteInputs
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.concat import Concat
+from mo.utils.error import Error
+from mo.ops.const import Const
+from mo.ops.op import PermuteAttrs
+
+
+class StridedSliceNormalizer(MiddleReplacementPattern):
+    """
+    StridedSlice is not normal if it cannot be permuted by ApplyPermutations. This normalizer
+    inserts blank colons ':' in slice expression so that it can be correctly permuted
+    from NHWC to NCHW layout. It changes masks and inserts blank begin, end and strides values.
+    In order to successfully handle StridedSlice in ShapeOf subgraphs
+    changes must be done by inserting nodes not just by overwriting constants.
+
+    StridedSlice is not normal in 2 cases:
+        1. rank of a slice expression is less than rank of input tensor
+        2. there is an ellipsis
+
+    1st case example
+    BEFORE:
+                  |
+                begin
+           value=[0, 0]
+                  |
+
+    AFTER:
+                  |
+                begin          Const
+           value=[0, 0]     value=[0, 0]
+                  \             /
+                   \           /
+                      Concat
+                 value=[0, 0, 0, 0]
+                        |
+
+    Input of a shape [16, 100, 100, 3] in NHWC layout, output = input[:, 0:50].
+    StridedSlice will be extended to input[:, 0:50, :, :].
+    After permutation to NCHW output = input[:, :, 0:50, :].
+    Example for 'begin' input transformation is shown above on the picture.
+    'end' and 'strides' inputs will be transformed the same way.
+
+    2nd case example
+    BEFORE:
+                  |
+                begin
+           value=[1, 50]
+                  |
+
+    AFTER:
+                  |
+                begin
+           value=[1, 1, 1]
+                  |
+             VariadicSplit
+           /              \
+          /                \
+         /       Const      \
+         \     val=[0, 0]   /
+          \       |        /
+           \      |       /
+               Concat
+           value=[1, 0, 0, 1, 1]
+                  |
+
+    Input of a shape [16, 10, 100, 100, 3] in NDHWC layout, output = input[1:4, ..., 1:51, 1:3],
+    output_shape = [3, 10, 100, 50, 2]. In order to perform correct layout permutation
+    ellipsis must be replaced with colons: input[1:4, ..., 1:51, 1:3] => input[1:4, :, :, 1:51, 1:3].
+    After layour permutation input[1:4, 1:3, :, : 1:5].
+
+    In the places of colons blank begin, end and strides values should be inserted.
+    In order to do that we split input and insert blank zeros to the middle.
+    Example for 'begin' input transformation is shown above on the picture.
+    'end' and 'strides' inputs will be transformed the same way.
+    """
+    enabled = True
```

Comment:
why do we perform this for all the frameworks and layouts?

Reply:
We need to specify 'slices' attribute which is used by `ConvertGroupedStridedSLices`. For this attribute similar "ellipsis unrolling" and "inputs extending" need to be done in `normalize_slices_attr`
https://github.com/openvinotoolkit/openvino/blob/a8da55a03a196414906ae03af86a8698337f2925/model-optimizer/extensions/middle/StridedSliceNormalizer.py#L219

If at some point we will get up to the removing `ConvertGroupedStridedSLices` we can add here `graph_condition` but even we don't this will not cause any harm.